src/main/webapp/scripts/poll.js

```diff
@@ -0,0 +1,65 @@
+ /** 
+  * The Poll class checks for fresh data over a given interval by
+  * periodically making API requests to a server. The Poll class can also
+  * be used to periodically update the server of necessary updates. 
+  * By default, polling objects poll indefinitely over a
+  * period of 30 seconds.
+  */ 
+ class Poll {
```

Comment:
Isn't this class an abstraction of a real world phenomenon? Polling is a recognized technique I guess, here it would just be implemented in Javascript